src/main/java/hudson/plugins/ec2/win/WinConnection.java

```diff
@@ -76,8 +79,12 @@ public WindowsProcess execute(String commandLine, int timeout) {
     }
 
     private DiskShare getSmbShare(String path) throws IOException {
-        Connection connection = smbclient.connect(host);
-        Session session = connection.authenticate(authentication);
+        if(this.connection == null) {
+            this.connection = smbclient.connect(host);
+        }
+        if(this.session == null) {
+            this.session = connection.authenticate(this.authentication);
+        }
```

Comment:
### First socket leak
Call to `getSmbShare` does not close connection and session.
Fix: created one global instance of `connection` and `session` and closed them explicitly.